### PR TITLE
Make text editor menu buttons keyboard accessible

### PR DESCRIPTION
**Pull request by kiwikern**
RISDEV-4432

### CODE REVIEW

File: frontend/src/components/input/TextEditor.vue [hunk 1]
```diff
@@ -141,7 +142,6 @@ const editor = new Editor({
     emit("updateValue", editor.getHTML())
   },
   onFocus: () => (hasFocus.value = true),
-  onBlur: () => (hasFocus.value = false),
```
Comment (kiwikern) [inline frontend/src/components/input/TextEditor.vue:144]: This was moved to line 454 `focusout` as we don't want to hide the buttons if the focus moves from the editor input to the menu bar.

**Comment by kiwikern** [comment]
Problem: When you have a list in the editor, shift+tab will remove the list. 🤔

File: frontend/src/components/input/TextEditor.vue [hunk 2]
```diff
@@ -405,17 +446,34 @@ const resizeObserver = new ResizeObserver((entries) => {
 </script>
 
 <template>
-  <div id="text-editor" class="editor bg-white" fluid>
+  <div
+    id="text-editor"
+    ref="editorElement"
+    class="editor bg-white"
+    fluid
+    @focusout="
```
Comment (anne-ds) [inline frontend/src/components/input/TextEditor.vue:454]: Maybe lets extract this into a method "handleFocusout"?

File: frontend/src/components/input/TextEditor.vue [hunk 3]
```diff
@@ -405,17 +446,34 @@ const resizeObserver = new ResizeObserver((entries) => {
 </script>
 
 <template>
-  <div id="text-editor" class="editor bg-white" fluid>
+  <div
+    id="text-editor"
+    ref="editorElement"
+    class="editor bg-white"
+    fluid
```
Comment (anne-ds) [inline frontend/src/components/input/TextEditor.vue:453]: "Attribute fluid is not allowed here"

File: frontend/src/components/input/TextEditor.vue [hunk 4]
```diff
@@ -424,6 +482,8 @@ const resizeObserver = new ResizeObserver((entries) => {
             v-for="(button, index) in fixButtons"
             :key="index"
             v-bind="button"
+            ref="fixButtonElements"
+            :tab-index="-1"
             @toggle="handleButtonClick"
           />
         </div>
```
Comment (kiwikern) [inline frontend/src/components/input/TextEditor.vue:489]: nb: We could extract this into a new TextEditorToolBar component

File: frontend/test/test-helper/prosemirror-document-mock.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+export const mockDocumentForProsemirror = () => {
```
Comment (kiwikern) [inline frontend/test/test-helper/prosemirror-document-mock.ts:3]: Add documentation + link to github source

**Pull request merged**